app/models/annotation/nml/NmlWriter.scala

```diff
@@ -190,4 +198,38 @@ object NmlWriter {
       }
     }
   }
+
+  def writeMetaData(annotation: Annotation)(implicit writer: XMLStreamWriter) = {
+    val userNameFuture = annotation.user.map(_.name).getOrElse("No user")
+    var userName = ""
+    userNameFuture.onComplete {
```

Comment:
What is a better way to do this?
I tried to handle it with a for-yield expression but then the Future/Fox would cause preoblems somewhere else

Reply:
just like with the taskId, I would suggest to not write the user tag if no user is present (instead of the "no user" string.
changing a var in an `onComplete` method is dangerous, since the onComplete callback is called asynchronously in a different thread than the surrounding code.
our options are 
a) to use Await (see https://docs.scala-lang.org/overviews/core/futures.html#blocking-outside-the-future )
b) to somehow pass the username into this function from outside where things are still asynchronous
c) to make the NMLWriter asynchronous

None of which is super easy, I fear. @jfrohnhofen any suggestions?